project_task_code/models/project_task.py

```diff
@@ -8,26 +8,18 @@ class ProjectTask(models.Model):
     _inherit = 'project.task'
 
     code = fields.Char(
-        string='Task Number', required=True, default="/", readonly=True)
+        string='Task Number',
+        required=True,
+        default=lambda self: self.env['ir.sequence'].next_by_code(
+            'project.task'),
+        readonly=False,
```

Comment:
After this change the code field could be changed in the form. Is this the desired behavior?
In the end: This should be no problem at all. Maybe `next_by_code` generation could fail if next free number is already taken manually.

If this is correct you could remove `readonly = False`. It's the default.

Reply:
Hello,
Thank you for your comments! 

I actually got caught in an incoherency between v9-10 (readonly=True) and v12 (readonly=False).
My attention was focused on the code error not the readonly property. Now that you mention it, I will give it a quick thought (and test)...

Reply:
>  I'll do this after checking project_key and project_task_code addons compatibility.
I think there is an overlap, indeed. `project_xxx_code` look simple, maybe a unifying work should be done. I came from `project_issue_code` which got lost in what I would call a regression of killing the `issue` in merging it into `task`. In my view two different thing with different uses...
For sure this PR solves this issue.

Reply:
As far as readonly=false is concerned, I got caught into the difference between v9-10 and v12 (RO=false was introduced with the commit 188e105418827a5fe785daae88d4a6f40ee3835d, for a reason I am not even sure holds true/is needed).

Now, especially as we also have "issues", one might want to change the code, from e.g. T0038 to I0038. So RW could be useful. However, the system won't let you save a duplicate code. The ir sequence mechanism is not perfect, but I see no big regression.

Reply:
Please have a look. I did the changes.